src/platform/baytrail/include/platform/memory.h

```diff
@@ -105,21 +105,23 @@ void platform_init_memmap(void);
  * +---------------------+----------------+-----------------------------------+
  */
 
-
 /* Heap section sizes for module pool */
 #define HEAP_RT_COUNT8		0
 #define HEAP_RT_COUNT16		48
 #define HEAP_RT_COUNT32		48
 #define HEAP_RT_COUNT64		32
 #define HEAP_RT_COUNT128	32
-#define HEAP_RT_COUNT256	32
```

Comment:
Why are you making block 256 count smaller ?

Reply:
@lgirdwood   This is the summary of allocation. So I decreased the size of 256. Do you perfer to change the block number of other size ?

````
841     heap: runtime status
815      heap: 0xff317080 size 17664 blocks 8 caps 0x61
817       used 8704 free 8960
825       block 1 base 0xff317380 size 32
828        count 48 free 48
825       block 2 base 0xff317980 size 64
828        count 32 free 32
825       block 3 base 0xff318180 size 128
828        count 16 free 3 
825       block 4 base 0xff318980 size 192
828        count 16 free 5 
825       block 5 base 0xff319580 size 256
828        count 24 free 8 
825       block 6 base 0xff31ad80 size 320
828        count 2 free 1  
825       block 7 base 0xff31b280 size 384
828        count 2 free 2
````

Reply:
@RanderWang why do you have blocks defined for 384 ? when this is the buffer block size ? i.e. it's not needed as code will use buffer heap here.

Reply:
@lgirdwood  we have got agreement at this 4 days ago, right ?
````
 lgirdwood 4 days ago Member

Why are you adding 384 when it's the buffer block size ?
@RanderWang
RanderWang 4 days ago •
Author Member

Now the algorithm is:

   if (size >= 384)
        ptr =  rballoc
   else
        ptr = _malloc

    if (!ptr)
        ptr = rballoc

Without 384 block, for the allocation 256 < size < 384, it can't be supported by _malloc but wastes
 cpu time in _malloc. So I add this to fix it
@lgirdwood
lgirdwood 4 days ago Member

There is opportunity now (since we have save memory by deleting 512 and 1024 blocks) to add 
more step sizes here. You could also add 320 block size. It does need a macro though for your 
largest block size before rballoc blocks for the logic statement above.

````

Reply:
@RanderWang I mean you dont need to add any blocks of size 384 to the non buffer heap map since all blocks of 384 size will be in buffer heap.

Reply:
@lgirdwood  yes, but I talk about allocation size (.eg 360 )< 384 and > 320. For this kind of allocation, It will be not allocated from buffer heap but it is not supported by other non buffer heap without 384 included.

Reply:
The idea is there is a gap between 320 and 384
````
---------------                        -----------------------
               320                     384
````

Reply:
ok, now I see to fill this gap, but it would be better to remove
```
#define HEAP_RT_LP_COUNT384		4
```
and catch all allocations above the largest block size (in this case 320) to us the buffer heap.

Reply:
I checked there is no free block of 384 sized on apollolake and whiskylake, icelake,  so I will remove 384 on Baytray and haswell only. Thanks!

Reply:
@RanderWang have you pushed the updated branch, I'm still seeing block size 384 for some heaps (not all)